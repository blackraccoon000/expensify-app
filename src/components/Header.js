import React from "react"
import { NavLink } from "react-router-dom"

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <ol>
      <li><NavLink activeClassName="is-active" to="/" exact>Dashboard</NavLink></li>
      <li><NavLink activeClassName="is-active" to="/create">Create Expense</NavLink></li>
      <li><NavLink activeClassName="is-active" to="/help">Help</NavLink></li>
      <li><NavLink activeClassName="is-active" to="/404">NotFound</NavLink></li>
    </ol>
    <hr/>
  </header>
)

export default Header