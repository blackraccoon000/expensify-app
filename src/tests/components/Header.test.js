import React from "react"
import { shallow } from "enzyme"
// import ReactShallowRenderer from "react-test-renderer/shallow"
import Header from "../../components/Header.js"

test("should render Header component correctly",() => {
  const wrapper = shallow(<Header/>)
  // expect(wrapper.find("h1").length).toBe(1)
  // expect(wrapper.find("h1").text()).toBe("Expensify")
  // console.log(wrapper.find("h1").text())
  expect(wrapper).toMatchSnapshot()
  // expect(toJSON(wrapper)).toMatchSnapshot()

  // const renderer = new ReactShallowRenderer
  // renderer.render(<Header />)
  // const result = renderer.getRenderOutput()
  // expect(result).toMatchSnapshot()
})