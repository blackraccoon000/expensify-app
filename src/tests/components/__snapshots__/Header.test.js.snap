// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <ol>
    <li>
      <NavLink
        activeClassName="is-active"
        aria-current="page"
        exact={true}
        to="/"
      >
        Dashboard
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        aria-current="page"
        to="/create"
      >
        Create Expense
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        aria-current="page"
        to="/help"
      >
        Help
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        aria-current="page"
        to="/404"
      >
        NotFound
      </NavLink>
    </li>
  </ol>
  <hr />
</header>
`;
